core/src/main/java/io/questdb/cairo/TableConverter.java

```diff
@@ -0,0 +1,148 @@
+import io.questdb.std.str.StringSink;
+
+import static io.questdb.cairo.TableUtils.*;
+import static io.questdb.cairo.wal.WalUtils.CONVERT_FILE_NAME;
+
+class TableConverter {
+    private static final Log LOG = LogFactory.getLog(TableConverter.class);
+
+    static void convertTables(CairoConfiguration configuration, TableSequencerAPI tableSequencerAPI) {
+        final Path path = Path.getThreadLocal(configuration.getRoot());
+        final int rootLen = path.length();
+        final StringSink sink = Misc.getThreadLocalBuilder();
+        final FilesFacade ff = configuration.getFilesFacade();
+        final long findPtr = ff.findFirst(path.$());
+        do {
+            if (ff.isDirOrSoftLinkDirNoDots(path, rootLen, ff.findName(findPtr), ff.findType(findPtr), sink)) {
+                if (!ff.exists(path.concat(WalUtils.CONVERT_FILE_NAME).$())) {
+                    continue;
+                }
+                final String dirName = sink.toString();
+                final boolean walEnabled = readWalEnabled(path, ff);
+                LOG.info().$("Converting table [dirName=").$(dirName).$(", walEnabled=").$(walEnabled).$(']').$();
+
+                path.trimTo(rootLen).concat(dirName);
+                try (final MemoryMARW metaMem = Vm.getMARWInstance()) {
+                    openSmallFile(ff, path, rootLen, metaMem, META_FILE_NAME, MemoryTag.MMAP_SEQUENCER_METADATA);
+                    if (metaMem.getBool(TableUtils.META_OFFSET_WAL_ENABLED) == walEnabled) {
+                        LOG.info().$("Skipping conversion, table already has the expected type [dirName=").$(dirName).$(", walEnabled=").$(walEnabled).$(']').$();
+                    } else {
+                        if (walEnabled) {
+                            final String tableName;
+                            try (final MemoryCMR mem = Vm.getCMRInstance()) {
+                                final String name = TableUtils.readTableName(path, mem, ff);
+                                tableName = name != null ? name : dirName;
+                            }
+
+                            final int tableId = metaMem.getInt(TableUtils.META_OFFSET_TABLE_ID);
+                            final TableToken token = new TableToken(tableName, dirName, tableId, true);
+                            try (TableReaderMetadata metadata = new TableReaderMetadata(configuration, token)) {
+                                metadata.load();
+                                tableSequencerAPI.registerTable(tableId, new TableDescriptorImpl(metadata), token);
+                            }
+                        }
+                        metaMem.putBool(TableUtils.META_OFFSET_WAL_ENABLED, walEnabled);
+                    }
+
+                    path.trimTo(rootLen).concat(dirName).concat(CONVERT_FILE_NAME);
+                    if (!ff.remove(path)) {
+                        LOG.error().$("Could not remove _convert file [path=").$(path).$(']').$();
+                    }
+                }
```

Comment:
Catch and LOG critical path with the failure table path.

Reply:
plus path as `.utf8` , and ... -> `.I$()` for extra care

Reply:
fixed in [277ba3c](https://github.com/questdb/questdb/pull/2947/commits/277ba3cad950ce380465d21517d4e1a40f82fbe2)